src/Spectre.AngularClient/src/app/heatmaps/shared/heatmap.service.ts

```diff
@@ -40,39 +40,48 @@ export class HeatmapService extends Service {
       console.log('[HeatmapService] preparationId: ' + preparationId + ' channelId ' + channelId);
       const queryUrl = `${this.getBaseUrl()}/heatmap/${preparationId}?channelId=${channelId}&flag=false`;
       const response = this.http.get(queryUrl, {headers: this.getHeaders()});
-      const spectrum = response.map(toHeatmap);
+      const spectrum = response.map((res: Response) => HeatmapUtil.toHeatmap(res));
       return spectrum;
     }
 }
 
-function toHeatmap(response: Response): Heatmap {
-    console.log('[HeatmapService] parsing response:');
-    console.log(response);
+export class HeatmapUtil {
+  static toHeatmap(response: Response, className = '[HeatmapService]'): Heatmap {
+    console.debug(className + ' parsing response:');
+    console.debug(response);
     const json = response.json();
-    console.log(json);
+    console.debug(json);
     const max_column = Math.max.apply(null, json.X);
     const min_column = Math.min.apply(null, json.X);
     const max_row = Math.max.apply(null, json.Y);
     const min_row = Math.min.apply(null, json.Y);
-    console.log('[HeatmapService] found bounds: (' + min_row + ', ' + max_row + '), (' + min_column + ', ' + max_column + ')');
-    console.log('[HeatmapService] initializing heatmap');
+    console.debug(className + ' found bounds: (' + min_row + ', ' + max_row + '), (' + min_column + ', ' + max_column + ')');
+    console.debug(className + ' initializing heatmap');
     const data = [];
     for (let i = min_row; i < max_row + 1; i++) {
       data[i - min_row] = [];
       for (let j = min_column; j < max_column + 1; j++) {
         data[i - min_row][j - min_column] = 0;
       }
     }
-    console.log('[HeatmapService] processing heatmap');
+    console.debug(className + ' processing heatmap');
     for (let i = 0; i < json.X.length; i++) {
-        const column = json.X[i] - min_column;
-        const row = max_row - json.Y[i];
+      const column = json.X[i] - min_column;
+      const row = max_row - json.Y[i];
+      if (className === '[HeatmapService]') {
```

Comment:
What is the difference between both cases?

Reply:
It should be normalized for both cases.

Reply:
On backend? It can be strange in heatmap have data instead of intensity field